precompile_process.go

```diff
@@ -43,11 +43,35 @@ func (p PrecompileProcess) Execute(workingDir string) error {
 		Args:   args,
 		Stdout: p.logger.ActionWriter,
 		Stderr: p.logger.ActionWriter,
-		Env:    append(os.Environ(), "RAILS_ENV=production", "SECRET_KEY_BASE=dummy"),
+		Env:    processPrecompileEnv(os.Environ()),
 	})
 	if err != nil {
 		return fmt.Errorf("failed to execute bundle exec output:\n%s\nerror: %s", buffer.String(), err)
 	}
 
 	return nil
 }
+
+func processPrecompileEnv(env []string) []string {
```

Comment:
This would definitely work but I think we could simplify this code a little bit by getting rid of the boolean variables and just checking if the env does not contain the prefix, like
```
if !strings.HasPrefix(pair, "RAILS_ENV=") {
  env = append(env, "RAILS_ENV=production")
}
```

Reply:
We have to check every pair in the env, not just the first one, so I don't think that would work? `strings.HasPrefix()` takes a single argument, not a slice of strings.

Reply:
Oh yea, you're so right, total brain fart there. We do need to loop over all of them, or find a nice way to leverage `os.LookupEnv`

Reply:
Either way is fine with me

Reply:
The unfortunate thing is that `os.LookupEnv` doesnt support an arbitrary map, so if we want to separate the logic here regarding os parsing, then we can't use that. That said, I don't mind either way, so let me know if you wish for me to update this.

Reply:
Yea, I hear you. I think I'm fine with this option